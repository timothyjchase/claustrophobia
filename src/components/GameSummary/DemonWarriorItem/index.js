import React from 'react'
import PropTypes from 'prop-types'
import renderHTML from 'react-render-html'
import { Button, Label, List, Popup, Rating } from 'semantic-ui-react'

const DemonWarriorItem = ({ warrior }) => {
  if (warrior.numberInPlay)
    return (
      <List.Item>
        <List.Content>
          <Button
            icon="remove"
            floated="right"
            size="tiny"
            onClick={warrior.onRemove}
            compact
            basic
          />
          <List.Header as="a">
            <Label circular horizontal>
              {warrior.numberInPlay}
            </Label>
            <Popup
              inverted
              trigger={(
                <span>
                  {`${warrior.name}${warrior.numberInPlay > 1 ? 's' : ''}`}
                </span>
)}
            >
              <div>
                <Label content={`${warrior.movement} MVT`} />
                {' '}
                <Label content={`${warrior.combat} CBT`} />
                {' '}
                <Label content={`${warrior.defense} DEF`} />
                {!!warrior.rules && <div>{renderHTML(warrior.rules)}</div>}
              </div>
            </Popup>
            {warrior.health > 1 && (
              <div style={{ paddingLeft: '30px' }}>
                <Rating
                  icon="heart"
                  defaultRating={0}
                  maxRating={warrior.health}
                  size="tiny"
                />
              </div>
            )}
          </List.Header>
        </List.Content>
      </List.Item>
    )
  return null
}

DemonWarriorItem.propTypes = {
  warrior: PropTypes.object.isRequired,
}

export default DemonWarriorItem
