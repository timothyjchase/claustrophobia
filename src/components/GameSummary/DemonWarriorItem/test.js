import React from 'react'
import DemonWarriorItem from './index'

describe('<DemonWarriorItem />', () => {
  describe('with no warrior', () => {
    const component = shallow(
      <DemonWarriorItem
        warrior={{
          numberInPlay: 0,
          name: 'Test warrior',
          movement: 1,
          combat: 2,
          defense: 4,
          health: 5,
          rules: '',
          onRemove: jest.fn(),
        }}
      />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with warrior, number in play=1, health = 5, rules', () => {
    const component = shallow(
      <DemonWarriorItem
        warrior={{
          numberInPlay: 1,
          name: 'Test warrior',
          movement: 1,
          combat: 2,
          defense: 4,
          health: 5,
          rules: 'custom',
          onRemove: jest.fn(),
        }}
      />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with warrior, number in play=2, health = 1, no rules', () => {
    const component = shallow(
      <DemonWarriorItem
        warrior={{
          numberInPlay: 2,
          name: 'Test warrior',
          movement: 1,
          combat: 2,
          defense: 4,
          health: 1,
          onRemove: jest.fn(),
        }}
      />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
