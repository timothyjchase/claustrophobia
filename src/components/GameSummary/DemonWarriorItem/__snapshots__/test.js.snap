// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DemonWarriorItem /> with no warrior should match snapshot 1`] = `""`;

exports[`<DemonWarriorItem /> with warrior, number in play=1, health = 5, rules should match snapshot 1`] = `
<ListItem>
  <ListContent>
    <Button
      as="button"
      basic={true}
      compact={true}
      floated="right"
      icon="remove"
      onClick={[MockFunction]}
      role="button"
      size="tiny"
    />
    <ListHeader
      as="a"
    >
      <Label
        circular={true}
        horizontal={true}
      >
        1
      </Label>
      <Popup
        inverted={true}
        keepInViewPort={true}
        on="hover"
        position="top left"
        trigger={
          <span>
            Test warrior
          </span>
        }
      >
        <div>
          <Label
            content="1 MVT"
          />
           
          <Label
            content="2 CBT"
          />
           
          <Label
            content="4 DEF"
          />
          <div>
            custom
          </div>
        </div>
      </Popup>
      <div
        style={
          Object {
            "paddingLeft": "30px",
          }
        }
      >
        <Rating
          clearable="auto"
          defaultRating={0}
          icon="heart"
          maxRating={5}
          size="tiny"
        />
      </div>
    </ListHeader>
  </ListContent>
</ListItem>
`;

exports[`<DemonWarriorItem /> with warrior, number in play=2, health = 1, no rules should match snapshot 1`] = `
<ListItem>
  <ListContent>
    <Button
      as="button"
      basic={true}
      compact={true}
      floated="right"
      icon="remove"
      onClick={[MockFunction]}
      role="button"
      size="tiny"
    />
    <ListHeader
      as="a"
    >
      <Label
        circular={true}
        horizontal={true}
      >
        2
      </Label>
      <Popup
        inverted={true}
        keepInViewPort={true}
        on="hover"
        position="top left"
        trigger={
          <span>
            Test warriors
          </span>
        }
      >
        <div>
          <Label
            content="1 MVT"
          />
           
          <Label
            content="2 CBT"
          />
           
          <Label
            content="4 DEF"
          />
        </div>
      </Popup>
    </ListHeader>
  </ListContent>
</ListItem>
`;
